File: app/src/main/java/com/example/dynamicviewpager/MainActivity.java
package com.example.dynamicviewpager;

import android.os.Bundle;
import android.os.Handler;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import java.util.ArrayList;
import java.util.Timer;
import java.util.TimerTask;

public class MainActivity extends AppCompatActivity {

    private ViewPager pager = null;
    private MainPagerAdapter pagerAdapter = null;
    ArrayList<String> list = new ArrayList<>();
    private int currentPage = 0;

    /**
     * https://homepages.cae.wisc.edu/~ece533/images/airplane.png
     * https://homepages.cae.wisc.edu/~ece533/images/arctichare.png
     * https://homepages.cae.wisc.edu/~ece533/images/baboon.png
     * https://homepages.cae.wisc.edu/~ece533/images/barbara.bmp
     * https://homepages.cae.wisc.edu/~ece533/images/boat.png
     */


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        list.add("https://homepages.cae.wisc.edu/~ece533/images/airplane.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/arctichare.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/baboon.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/barbara.bmp");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/boat.png");

        pagerAdapter = new MainPagerAdapter(getApplicationContext(), list);
        pager = findViewById(R.id.view_pager);
        pager.setAdapter(pagerAdapter);

        for (int i = 0; i < list.size(); i++) {
            pagerAdapter.addView(getLayoutInflater().inflate(R.layout.item_advertise, null), i);
        }

        pagerAdapter.notifyDataSetChanged();

        TabLayout tabLayout = findViewById(R.id.tabDots);
        tabLayout.setupWithViewPager(pager, true);

        setTimer();
    }

    private void setTimer() {

        final Handler handler = new Handler();
        final Runnable Update = new Runnable() {
            public void run() {
                if (currentPage == list.size()) {
                    currentPage = 0;
                }
                pager.setCurrentItem(currentPage++, true);
            }
        };
        Timer timer = new Timer(); // This will create a new Thread
        timer.schedule(new TimerTask() { // task to be scheduled
            @Override
            public void run() {
                handler.post(Update);
            }
        }, 500, 1800);
    }

    public void addView(View newPage) {
        int pageIndex = pagerAdapter.addView(newPage);
        // You might want to make "newPage" the currently displayed page:
        pager.setCurrentItem(pageIndex, true);
    }

    public void removeView(View defunctPage) {
        int pageIndex = pagerAdapter.removeView(pager, defunctPage);
        // You might want to choose what page to display, if the current page was "defunctPage".
        if (pageIndex == pagerAdapter.getCount())
            pageIndex--;
        pager.setCurrentItem(pageIndex);
    }

    public View getCurrentPage() {
        return pagerAdapter.getView(pager.getCurrentItem());
    }

    public void setCurrentPage(View pageToShow) {
        pager.setCurrentItem(pagerAdapter.getItemPosition(pageToShow), true);
    }

}

File: viewpager2/src/main/java/com/vinrak/viewpager2/MainActivity.kt
package com.vinrak.viewpager2

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val currentPage = intArrayOf(0)

        val viewpager2Test: ViewPager2 = findViewById(R.id.viewpager2Test)
        val list = arrayListOf("A", "B", "C")
        viewpager2Test.adapter = Pager2Adapter(list, applicationContext)

        val tabDots: TabLayout = findViewById(R.id.tabDots)

        TabLayoutMediator(tabDots, viewpager2Test, TabLayoutMediator.TabConfigurationStrategy { tab, position -> // Styling each tab here
            //tab.text = "Tab $position"
        }).attach()


        val handler = Handler()
        val update = Runnable {
            if (currentPage[0] == list.size) {
                currentPage[0] = 0
            }
            viewpager2Test.setCurrentItem(currentPage[0]++, true)
        }
        val timer = Timer() // This will create a new Thread
        timer.schedule(object : TimerTask() {
            // task to be scheduled
            override fun run() {
                handler.post(update)
            }
        }, 5000, 5000)

    }
}
